scipy/sparse/tests/test_sputils.py

```diff
@@ -83,12 +83,68 @@ def test_isdense(self):
         assert_equal(sputils.isdense(np.matrix([1])), True)
 
     def test_validateaxis(self):
-        func = sputils.validateaxis
-
-        assert_raises(TypeError, func, (0, 1))
-        assert_raises(TypeError, func, 1.5)
-        assert_raises(ValueError, func, 3)
+        assert_raises(TypeError, sputils.validateaxis, (0, 1))
+        assert_raises(TypeError, sputils.validateaxis, 1.5)
+        assert_raises(ValueError, sputils.validateaxis, 3)
 
         # These function calls should not raise errors
         for axis in (-2, -1, 0, 1, None):
-            func(axis)
+            sputils.validateaxis(axis)
+
+    def test_get_index_dtype(self):
+        imax = np.iinfo(np.int32).max
+        too_big = imax + 1
+
+        # Check that uint32's with no values too large doesn't return
+        # int64
+        arr1 = np.ones(90, dtype='uint32')
+        arr2 = np.ones(90, dtype='uint32')
+        arrs = (arr1, arr2)
+        assert_equal(
+            sputils.get_index_dtype(arrs, check_contents=True),
+            np.int32
+        )
+
+        # Check that if we can not convert but all values are less than or
+        # equal to max that we can just convert to int32
+        arr1[-1] = imax
+        arrs = (arr1, arr2)
+        assert_equal(
+            sputils.get_index_dtype(arrs, check_contents=True),
+            np.int32
+        )
+
+        # Check that if it can not convert directly and the contents are
+        # too large that we return int64
+        arr1[-1] = too_big
+        arrs = (arr1, arr2)
+        assert_equal(
+            sputils.get_index_dtype(arrs, check_contents=True),
+            np.int64
+        )
+
+        # test that if can not convert and didn't specify to check_contents
+        # we return int64
+        arr1 = np.ones(89, dtype='uint32')
+        arr2 = np.ones(89, dtype='uint32')
+        arrs = (arr1, arr2)
+        assert_equal(
+            sputils.get_index_dtype(arrs),
+            np.int64
+        )
+
+        # Check that even if we have arrays that can be converted directly
+        # that if we specify a maxval directly it takes precedence
+        arr1 = np.ones(12, dtype='uint32')
+        arr2 = np.ones(12, dtype='uint32')
+        arrs = (arr1, arr2)
+        assert_equal(
+            sputils.get_index_dtype(arrs, maxval=too_big, check_contents=True),
+            np.int64
+        )
```

Comment:
From what I can tell (possibly incorrect), `assert_equal` checks `==` equalty. For type comparisons, you might use `assert sputils.get_index_dtype(arrs, maxval=too_big, check_contents=True) is np.int64`.

Reply:
@xoviat The return value from sputils.get_index_dtype is np.int64 or np.int32 there are no other return values from the function. Since it is from np I think assert_equals is appropriate, no?

Reply:
I see what you mean from the logs in the test failure. Will change this and resubmit.

Reply:
I'm sorry, I think I had is right the first time. I would try `assert sputils.get_index_dtype(arrs, maxval=too_big, check_contents=True) is np.int64`.

Reply:
This may pose an issue though @xoviat 
```
>>> type(type(int)) == type(np.int32)
True
>>> type(type(int)) == type(np.int64)
True
>>> type(type(np.int32)) == type(np.int64)
True
```

Reply:
Yes, I first had the correct version and then changed it because I was unsure.

Reply:
Try it in the REPL first before wasting another hour on the CI.